### PR TITLE
Fix stale keys deleted metrics

### PR DESCRIPTION
**Pull request by yontyon**
In this PR we fix the metrics for keys deletions.
On deleteGenesisBlock we count only the deletions of stale keys

### CODE REVIEW

File: kvbc/include/categorization/block_merkle_category.h [hunk 1]
```diff
@@ -72,7 +72,7 @@ class BlockMerkleCategory {
 
   // Delete the given block ID as a genesis one.
   // Precondition: The given block ID must be the genesis one.
-  void deleteGenesisBlock(BlockId, const BlockMerkleOutput&, storage::rocksdb::NativeWriteBatch&);
```
Comment (dartdart26) [inline kvbc/include/categorization/block_merkle_category.h:75]: Can we put a comment that these functions return the number of keys actually deleted from disk? Applies to all categories.

Comment (dartdart26) [reply kvbc/include/categorization/block_merkle_category.h:75]: Nit: Can use `std::size_t` instead.

File: kvbc/src/categorization/block_merkle_category.cpp [hunk 2]
```diff
@@ -541,11 +544,17 @@ std::pair<SetOfKeyValuePairs, KeysVector> BlockMerkleCategory::rewriteAlreadyPru
...
 
-void BlockMerkleCategory::deleteGenesisBlock(BlockId block_id, const BlockMerkleOutput& out, NativeWriteBatch& batch) {
+size_t BlockMerkleCategory::deleteGenesisBlock(BlockId block_id,
+                                               const BlockMerkleOutput& out,
+                                               NativeWriteBatch& batch) {
   auto [hashed_keys, latest_versions] = getLatestVersions(out);
   auto overwritten_active_keys_from_pruned_blocks = findActiveKeysFromPrunedBlocks(hashed_keys);
+  size_t num_of_deletes = 0;
+  for (auto& kv : overwritten_active_keys_from_pruned_blocks) {
+    num_of_deletes += kv.second.size();
```
Comment (dartdart26) [inline kvbc/src/categorization/block_merkle_category.cpp:554]: Are we counting key deletes from disk only or stale data too?

Comment (yontyon) [reply kvbc/src/categorization/block_merkle_category.cpp:554]: stale only

Comment (dartdart26) [reply kvbc/src/categorization/block_merkle_category.cpp:554]: @andrewjstone Could you please also have a look at the block merkle category?

**Review by dartdart26 — APPROVED** [review]
Thanks @yontyon, looks good!

File: kvbc/src/categorization/block_merkle_category.cpp (hunk 2 above)
Comment (andrewjstone) [reply kvbc/src/categorization/block_merkle_category.cpp:554]: Looks fine to me. It's somewhat strange that we have 3 ways to return data, but 🤷

**Review by andrewjstone — APPROVED** [review]
Thanks for adding the support for this @yontyon

**Pull request merged**